codegen/smithy-aws-swift-codegen/src/test/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSDeprecatedShapeRemoverTests.kt

```diff
@@ -0,0 +1,65 @@
+package software.amazon.smithy.aws.swift.codegen.model
+
+import io.kotest.matchers.string.shouldContainOnlyOnce
+import io.kotest.matchers.string.shouldNotContain
+import org.junit.jupiter.api.Test
+import software.amazon.smithy.aws.swift.codegen.TestContext
+import software.amazon.smithy.aws.swift.codegen.TestUtils
+import software.amazon.smithy.aws.swift.codegen.shouldSyntacticSanityCheck
+import software.amazon.smithy.aws.traits.protocols.RestJson1Trait
+
+class AWSDeprecatedShapeRemoverTests {
+    @Test
+    fun `Shape deprecated before the cutoff date based on deprecated trait's since field gets removed`() {
+        val context = setupTests("deprecated-shape-removal-test.smithy", "com.test#Example")
+        val contents = TestUtils.getModelFileContents("Example", "OperationWithDeprecatedInputMembersInput.swift", context.manifest)
+        contents.shouldSyntacticSanityCheck()
+        val removedContent = """
+    @available(*, deprecated, message: " API deprecated since 2024-09-01")
```

Comment:
Why is there a leading space in the message?  Even if not part of this PR, we should remove that

Reply:
Good catch; fixed as part of https://github.com/smithy-lang/smithy-swift/pull/732.